### PR TITLE
Add tests for stringification_inspector

### PR DESCRIPTION
**Pull request by shariarriday**

### CODE REVIEW

**Comment by shariarriday** [comment]
While writing the test for this file I found the following flag does not seem to have any use currently. https://github.com/actor-framework/actor-framework/blob/a4b58cac35aab8fa3df5026259fdf21b0c2211ef/libcaf_core/caf/detail/stringification_inspector.hpp#L42. It may be a good idea to remove it and do necessary refactoring if it is not required.

**Comment by Neverlord** [comment]
> It may be a good idea to remove it and do necessary refactoring if it is not required.

ACK. I've made a note in #1873. Removing member variables breaks the ABI, so we can't do that change in a minor release.

File: libcaf_core/caf/detail/stringification_inspector.test.cpp [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+           R"([["hello", "world"]])"s);
+}
+
+TEST("stringification of std::chrono::duration") {
+  check_eq(stringification_inspector::render(std::chrono::seconds{42}), "42s"s);
+  check_eq(stringification_inspector::render(std::chrono::milliseconds{42}),
+           "42ms"s);
+  check_eq(stringification_inspector::render(std::chrono::microseconds{42}),
+           "42us"s);
+  check_eq(stringification_inspector::render(std::chrono::nanoseconds{42}),
```
Comment (Neverlord) [inline libcaf_core/caf/detail/stringification_inspector.test.cpp:102]:
```suggestion
  check_eq(stringification_inspector::render(42s), "42s"s);
  check_eq(stringification_inspector::render(42ms), "42ms"s);
  check_eq(stringification_inspector::render(42us), "42us"s);
  check_eq(stringification_inspector::render(42ns),
```

File: libcaf_core/caf/detail/stringification_inspector.test.cpp [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+  check_eq(stringification_inspector::render(timespan{42}), "42ns"s);
+  check_eq(stringification_inspector::render(timespan{42 * 1000}), "42us"s);
+  check_eq(stringification_inspector::render(timespan{42 * 1000 * 1000}),
+           "42ms"s);
+}
+
+TEST("stringification of byte span") {
+  check_eq(stringification_inspector::render(std::byte{42}), "2A"s);
+  check_eq(stringification_inspector::render(caf::span<const std::byte>{
+             reinterpret_cast<const std::byte*>("hello"), 5}),
```
Comment (Neverlord) [inline libcaf_core/caf/detail/stringification_inspector.test.cpp:119]:
```suggestion
  auto hello = "hello"sv;
  check_eq(stringification_inspector::render(as_bytes(make_span(hello))),
```

**Review by Neverlord — CHANGES_REQUESTED** [review]
Just a couple minor tweaks for readability.

**Review by Neverlord — APPROVED** [review]

**Pull request merged**